### DISCUSSION TITLE
Correct syntax for multiple filter items based on multiple T/F key press data points?

### DISCUSSION THREAD

**Discussion by edward-christian** [jspsych/jsPsych#920] [Q&A] [answered]
I put images in an array, here's what one such item looks like:

```
{ stimulus: 'img/picture.png', data: {test_part: 'test', correct_response: '3', alt_response: '4', alt2_response: '5'}},
```

In one of my experiments, I'm having it record the key-press data this way for each image shown from the array:

```
on_finish: function(data) {
  data.correct = data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode(data.correct_response);
  data.alt = data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode(data.alt_response);
  data.alt2 = data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode(data.alt2_response);
},
```

And I'm trying to get that data in a function this way to use it in another part of the experiment:
```
var trueImages = jsPsych.data.get().filter({correct: true, alt: true, alt2: true}).select('stimulus').values;
```

It doesn't work, comes back undefined. Where did my syntax or logic fail?

**Comment by edward-christian** [comment]
Note - when I change the function to filter and get data to:

```
var trueImages = jsPsych.data.get().filter({correct: true}).select('stimulus').values;
```

The experiment works perfectly again, and I can see that the data is capturing T/F for correct, alt, and alt2.

I'm obviously filtering wrong.

So --

How can I get get data and grab images that have TRUE for correct, alt and alt2?

**Reply by vijaymarupudi** [reply]
Try the regular Javascript way of doing this.

```javascript
jsPsych.data
  .get()
  .values()
  .filter((x) => x.correct && x.alt && x.alt2)
  .map((x) => x.stimulus);
```

**Comment by jodeleeuw** [comment]
Your data has fields for `alt_response` and `alt2_response`:

```js
{ stimulus: 'img/picture.png', data: {test_part: 'test', correct_response: '3', alt_response: '4', alt2_response: '5'}},
```

But you are filtering on `alt` and `alt2`:
```js
var trueImages = jsPsych.data.get().filter({correct: true, alt: true, alt2: true}).select('stimulus').values;
```

Fixing the name mismatch might solve it.

**Comment by edward-christian** [comment, marked as answer]
I figured it out...

Had to add brackets and filter like this:

```
var trueImages = jsPsych.data.get().filter({correct: true},{alt: true},{alt2: true}).select('stimulus').values;
```

**Reply by jodeleeuw** [reply]
Hmm, this is not the intended behavior of the method. See the [docs here](https://www.jspsych.org/core_library/jspsych-data/#filter). Are you sure the `alt` and `alt2` properties are being filtered like you expect here?
